src/aiida/engine/processes/process.py

```diff
@@ -415,8 +415,19 @@ def on_create(self) -> None:
     @override
     def on_entered(self, from_state: Optional[plumpy.process_states.State]) -> None:
         """After entering a new state, save a checkpoint and update the latest process state change timestamp."""
+        from plumpy import ProcessState
+
         from aiida.engine.utils import set_process_state_change_timestamp
 
+        super().on_entered(from_state)
+
+        if self._state.LABEL is ProcessState.EXCEPTED:
```

Comment:
Is there a reason we don't move this logic into the `try`-`except` below?

Reply:
Where exactly? The point is that if this conditional is true, we already excepted and we don't want to except again, because that would cause the bug we are fixing. I guess perhaps one other solution would be to wrap all custom code in this overridden method inside a try-except and only reraise if the current state is not already excepted.

Reply:
I was wondering why you don't add it after the `except ValueError` on line 439, only raising in case the process state is not already excepted. Then the process state would be set in the `finally` part anyways. I suppose in the current version, _all_ exceptions are pre-empted, but is this what we want?

Also, do the lines on 444-445 no longer have to be executed?

```python
        self._save_checkpoint()
        set_process_state_change_timestamp(self.node)
```

Reply:
My reasoning was that _if_ we are already excepted, we want to do the bare minimum necessary because any additional operation can cause another exception which would display the bug we are trying to fix where the node is not sealed. The updating of the checkpoint and process state change timestamp are not critical. So out of safety, I was just thinking to _only_ do the node state and then just wrap things up.